packages/panzer/disc-fe/src/lof/Panzer_BlockedEpetraLinearObjFactory_impl.hpp

```diff
@@ -1233,34 +1367,73 @@ buildColMap(int i) const
+//
+///////////////////////////////////////////////////////////////////////////////
+template <typename Traits, typename LocalOrdinalT>
+const Teuchos::RCP<Epetra_Map>
+BlockedEpetraLinearObjFactory<Traits, LocalOrdinalT>::
 buildGhostedMap(int i) const
 {
-   std::vector<int> indices;
-
-   // get the global indices
-   getGlobalIndexer(i)->getOwnedAndGhostedIndices(indices);
+  std::vector<int> indices;
+  getGlobalIndexer(i)->getOwnedAndGhostedIndices(indices);
+  return Teuchos::rcp(new Epetra_Map(-1, indices.size(), &indices[0], 0,
+    *eComm_));
+} // end of buildGhostedMap()
 
-   return Teuchos::rcp(new Epetra_Map(-1,indices.size(),&indices[0],0,*eComm_));
-}
+///////////////////////////////////////////////////////////////////////////////
+//
+//  buildGhostedMap2()
+//
+///////////////////////////////////////////////////////////////////////////////
+template <typename Traits, typename LocalOrdinalT>
+const Teuchos::RCP<Epetra_Map>
+BlockedEpetraLinearObjFactory<Traits, LocalOrdinalT>::
+buildGhostedMap2(int i) const
+{
+  std::vector<int> indices;
+  getGlobalIndexer(i)->getGhostedIndices(indices);
+  return Teuchos::rcp(new Epetra_Map(-1, indices.size(), &indices[0], 0,
+    *eComm_));
+} // end of buildGhostedMap2()
 
-// build the ghosted map
-template <typename Traits,typename LocalOrdinalT>
-const Teuchos::RCP<Epetra_Map> BlockedEpetraLinearObjFactory<Traits,LocalOrdinalT>::
+///////////////////////////////////////////////////////////////////////////////
+//
+//  buildColGhostedMap()
+//
+///////////////////////////////////////////////////////////////////////////////
+template <typename Traits, typename LocalOrdinalT>
+const Teuchos::RCP<Epetra_Map>
+BlockedEpetraLinearObjFactory<Traits, LocalOrdinalT>::
 buildColGhostedMap(int i) const
 {
-   if(not useColGidProviders_)
-     return buildGhostedMap(i);
-
-   std::vector<int> indices;
-
-   // get the global indices
-   getColGlobalIndexer(i)->getOwnedAndGhostedIndices(indices);
-
-   return Teuchos::rcp(new Epetra_Map(-1,indices.size(),&indices[0],0,*eComm_));
-}
+  if (not useColGidProviders_)
+    return buildGhostedMap(i);
+  std::vector<int> indices;
+  getColGlobalIndexer(i)->getOwnedAndGhostedIndices(indices);
+  return Teuchos::rcp(new Epetra_Map(-1, indices.size(), &indices[0], 0,
+    *eComm_));
+} // end of buildColGhostedMap()
+
+///////////////////////////////////////////////////////////////////////////////
+//
+//  buildColGhostedMap2()
+//
+///////////////////////////////////////////////////////////////////////////////
+template <typename Traits, typename LocalOrdinalT>
+const Teuchos::RCP<Epetra_Map>
+BlockedEpetraLinearObjFactory<Traits, LocalOrdinalT>::
+buildColGhostedMap2(int i) const
+{
+  if (not useColGidProviders_)
+    return buildGhostedMap2(i);
+  std::vector<int> indices;
+  getColGlobalIndexer(i)->getGhostedIndices(indices);
+  return Teuchos::rcp(new Epetra_Map(-1, indices.size(), &indices[0], 0,
+    *eComm_));
+} // end of buildColGhostedMap2()
```

Comment:
Same as original, but calls either `buildGhostedMap2`, or calls `getGhostedIndices` instead of `getOwnedAndGhostedIndices`.